rmf_traffic/src/rmf_traffic/schedule/Negotiation.cpp

```diff
@@ -186,13 +187,17 @@ class Negotiation::Table::Implementation
     const auto it = std::remove_if(unsubmitted.begin(), unsubmitted.end(),
                    [&](const ParticipantId p) { return p == participant; });
     assert(it != unsubmitted.end());
-    unsubmitted.erase(it);
+    unsubmitted.erase(it, unsubmitted.end());
```

Comment:
is there a reason we are erasing all the participants after the owner participant as well? Unless the order of `ParticipantId`s is meaningful in another way?

Reply:
The order of participants in `unsubmitted` isn't meaningful until line 187. At that point, every element in `submitted` from `it` through `end()` are equal to `participant`, because that's how `std::remove_if` works.

This is called the [Erase-remove Idiom](https://en.wikipedia.org/wiki/Erase%E2%80%93remove_idiom) in C++. It's an efficient way of clearing elements out of a vector that fulfill some criteria. In this case, the criteria is having a value that's equal to `participant`.

In theory the vector `unsubmitted` should be given to us without any duplicate entries, so we could be lazier and just do a find -> erase instead of a remove -> erase, but the remove -> erase pattern makes us more robust to any possible upstream errors.

Reply:
thanks for the explanation! never knew about this, will look into using it in the future